elastalert/elastalert.py

```diff
@@ -784,7 +784,10 @@ def alert(self, matches, rule, alert_time=None):
             try:
                 alert.alert(matches)
             except EAException as e:
-                self.handle_error('Error while running alert %s: %s' % (alert.get_info()['type'], e), {'rule': rule['name']})
+                self.handle_error('ElastAlert Error while running alert %s: %s' % (alert.get_info()['type'], e), {'rule': rule['name']})
+                alert_exception = str(e)
+            except Exception as e:
+                self.handle_error('Unexpected Error while running alert %s: %s' % (alert.get_info()['type'], e), {'rule': rule['name']})
```

Comment:
I think it would be nice to have some kind of smart exception handling, so that an unexpected exception can cause ElastAlert to disable the specific rule that threw that exception and continue with the rest. Almost all bugs we've found trigger every time a specific alert/rule runs, and this would cause the alert/rule to continually throw exceptions every time ElastAlert runs.